### PR TITLE
feat(addon create): choose the cheapest plan by default

### LINKED ISSUE #653: clever addon create: use cheapest/lowest plan by default
## Context

Right now `clever addon create` uses `dev` plan by default when not specified, it only works for PG and MySQL. We should use the cheapest/lowest plan by default.

### PR DESCRIPTION
**Pull request by pdesoyres-cc**
Fixes #653

### CODE REVIEW

**Review by aurrelhebert — APPROVED** [review]

**Comment by aurrelhebert** [comment]
QA ok :heavy_check_mark: it's working like a charm

File: src/models/addon.js [hunk 1]
```diff
@@ -289,6 +285,24 @@ function parseAddonOptions (options) {
   }, {});
 }
 
+async function getPlan (planName, provider) {
```
Comment (hsablonniere) [inline src/models/addon.js:288]: nitpick: this function does not need to be async

Comment (hsablonniere) [reply src/models/addon.js:288]: nitpick: we could pass the list of plans instead of the whole provider

File: src/models/addon.js [hunk 2]
```diff
@@ -289,6 +285,24 @@ function parseAddonOptions (options) {
...
+  // if no plan specified, pick the cheapest one
+  if (planName == null || planName === '') {
+    if (provider.plans.length === 0) {
+      throw new Error('no plans available');
```
Comment (hsablonniere) [inline src/models/addon.js:292]: question: this will be displayed to the user, is it descriptive enough?

File: src/models/addon.js [hunk 3]
```diff
@@ -289,6 +285,24 @@ function parseAddonOptions (options) {
...
+    }
+
+    return provider.plans.sort((p1, p2) => p1.price - p2.price)[0];
+  }
+
+  const plan = provider.plans.find((p) => p.slug.toLowerCase() === planName.toLowerCase());
+  if (plan == null) {
+    const availablePlans = provider.plans.map((p) => p.slug);
+    throw new Error(`invalid plan name. Available plans: ${availablePlans.join(', ')}`);
```
Comment (hsablonniere) [inline src/models/addon.js:301]: nitpick: we don't have copy guidelines but I would have used an upper case letter at the beginning and a coma before the "available plans:".

**Review by hsablonniere — CHANGES_REQUESTED** [review]
Well done @pdesoyres-cc :clap: 

Tested with the built version on:

* PG no plan
* PG with invalid plan
* PG with valid plan
* redis no plan
* kv no plan
* keycloak no plan
* cellar no plan
* config provider

I added a few comments here and there. I wonder if we should display (human and JSON) the selected plan since we may have selected it for the user.

File: src/models/addon.js (hunk 3 above)
Comment (pdesoyres-cc) [reply src/models/addon.js:301]: This was not the subject of this PR, but I can do that.

File: src/models/addon.js (hunk 2 above)
Comment (pdesoyres-cc) [reply src/models/addon.js:292]: This was not the subject of this PR, but I can do that.

Comment (pdesoyres-cc) [reply src/models/addon.js:292]: Do you have a more descriptive suggestion?

Comment (pdesoyres-cc) [reply src/models/addon.js:292]: I have introduced this if branch for code robustness. But, this should never happen if the API always returns plans for a available provider (which, as for now, is the case for every providers).

Comment (hsablonniere) [reply src/models/addon.js]: Let's discuss it together in a meeting and move on :wink:

File: src/models/addon.js (hunk 3 above)
Comment (hsablonniere) [reply src/models/addon.js]: Ah you're right, sorry. We'll discuss it in the meeting :+1:

File: src/models/addon.js [hunk 4]
```diff
@@ -136,17 +136,16 @@ function validateAddonVersionAndOptions (region, version, addonOptions, provider
...
   if (!provider.regions.includes(region)) {
-    throw new Error(`invalid region name. Available regions: ${provider.regions.join(', ')}`);
+    throw new Error(`Invalid region name. Available regions: ${provider.regions.join(', ')}`);
+  }
+  if (provider.plans.length === 0) {
+    throw new Error(`No plans available for provider ${providerName}`);
   }
 
-  const plan = await getPlan(planName, provider);
+  const plan = await getPlan(planName, provider.plans);
```
Comment (hsablonniere) [inline src/models/addon.js:148]: issue: you can remove the `await`

**Review by hsablonniere — CHANGES_REQUESTED** [review]

File: src/models/addon.js [hunk 5]
```diff
@@ -289,6 +284,20 @@ function parseAddonOptions (options) {
   }, {});
 }
 
+function getPlan (planName, plans) {
```
Comment (miton18) [inline src/models/addon.js:287]:
```suggestion
function getDefaultPlan (planName, plans) {
```

**Review by miton18 — APPROVED** [review]

**Review by hsablonniere — APPROVED** [review]

**Pull request merged**